### PR TITLE
Adding tests for users/databases/root.

### PR DESCRIPTION
**Pull request by hub-cap**
- Added skiptest for tests not implemented.
- Added wait_for_ip to enable the dependent tests.

### CODE REVIEW

**Comment by cp16net** [comment]
+1

File: tests/integration/tests/api/instances.py [hunk 1]
```diff
@@ -557,7 +559,7 @@ def setUp(self):
     @test
     def test_detail_list(self):
         expected_attrs = ['created', 'flavor', 'hostname', 'id', 'links',
-                          'name', 'status', 'updated', 'volume']
+                          'name', 'status', 'updated', 'volume', 'ip']
```
Comment (rnirmal) [inline tests/integration/tests/api/instances.py:562]: Is the ip going to be just {'ip': ''} or {'addresses': [{'ip': ''}, {'ip': ''}]}

File: tests/integration/tests/openvz/dbaas_ovz.py [hunk 2]
```diff
@@ -51,15 +54,31 @@ class TestMultiNic(object):
...
+            return False
+        poll_until(get_ip_for_instance, sleep_time=5, time_out=20)
+
+    @test
     def test_multi_nic(self):
         """
         Multinic - Verify that nics as specified in the database are created
         in the guest
         """
+        if test_config.values['openvz_disabled']:
```
Comment (TimSimpsonR) [inline tests/integration/tests/openvz/dbaas_ovz.py]: This checks nics in the actual DB, so it could just use:

<pre><code>
if WHITEBOX:
</code></pre>

Comment (hub-cap) [inline tests/integration/tests/openvz/dbaas_ovz.py]: It does? I thought it used te vz stuff to find the ip.

Comment (TimSimpsonR) [inline tests/integration/tests/openvz/dbaas_ovz.py]: Hmm, maybe all these tests could have their enabled property set by "not test_config.values['openvz_disabled'] given the filename. But my point was since its a white box test you could guard it with that var too.

File: scripts/conf/test.conf [hunk 3]
```diff
@@ -69,5 +69,10 @@
           }
         }
     ],
+    "root_removed_from_instance_api": true,
+    "root_timestamp_disabled": true,
+    "openvz_disabled": true,
+    "management_api_disabled": true,
```
Comment (TimSimpsonR) [inline scripts/conf/test.conf:75]: This is the inverse of "test_mgmt".
In general I think we should stick to whether things are enabled versus whether they are disabled, since its easier to write those into the "enabled" property or if statements.

File: tests/integration/tests/api/root.py [hunk 4]
```diff
@@ -113,6 +128,8 @@ def test_root_now_enabled(self):
     @test(depends_on=[test_root_now_enabled])
     def test_root_now_enabled_details(self):
         """Use instance details to test that root is now enabled."""
+        if test_config.values['root_removed_from_instance_api']:
+            raise SkipTest("Root is no longer in the instances api")
```
Comment (TimSimpsonR) [inline tests/integration/tests/api/root.py]: Raising SkipTest for things we plan on adding back soon keeps us from ignoring the tests long term, but I think if the tests aren't being run because they don't apply to a configuration (as in this case, where we may never run this test again) the @test "enabled" property should be used, i.e.

<pre><code>
 @test(depends_on=[test_root_now_enabled], enabled=not test_config.values['root_removed_from_instance_api'])
</code></pre>

This is also why I think the test conf values will be easier to deal with if they don't state a negative, i.e.

<pre><code>
 @test(depends_on=[test_root_now_enabled], enabled=test_config.values['instance_api_has_root'])
</code></pre>

**Comment by TimSimpsonR** [comment]
+1

**Comment by ed-** [comment]
+1
